compiler/src/dotty/tools/dotc/transform/ParamForwarding.scala

```diff
@@ -66,8 +68,12 @@ class ParamForwarding(thisTransformer: DenotTransformer) {
                     sym.copySymDenotation(initFlags = sym.flags | Method | Stable, info = sym.info.ensureMethodic)
                       .installAfter(thisTransformer)
                     val superAcc =
-                      Super(This(currentClass), tpnme.EMPTY, inConstrCall = false).select(alias)
-                    DefDef(sym, superAcc.ensureConforms(sym.info.widen))
+                      Super(This(currentClass), tpnme.EMPTY, inConstrCall = false)
+                        .select(alias)
+                    val stpe @ TermRef(_, _) = superAcc.tpe
+                    val superAccShadowed = superAcc.withType(stpe.shadowed)
```

Comment:
Technically, this is only necessary when the super is not the base class where the paramaccessor is defined, but I guess there's no harm in always using `.shadowed`. However it would be nice to add a comment explaining when the `.shadowed` is useful.